tests/unit/metrics/domain/common/test_utils.py

```
@@ -344,3 +354,106 @@ def test_returns_true_for_vaccination_testing_substrings(self, metric_name: str)
+        Then the expected response is returned
+        """
+        # Given
+        expected_metric_group = expected_return_value
+
+        # When
+        returned_metric_group = DataSourceFileType[metric_group].is_headline
+
+        # Then
+        assert returned_metric_group == expected_metric_group
+
+    @pytest.mark.parametrize(
+        "metric_group, expected_return_value",
+        (
+            [
+                ("headline", False),
+                ("cases", True),
+                ("deaths", True),
+                ("testing", True),
+                ("healthcare", True),
+                ("vaccinations", True),
+            ]
+        ),
+    )
+    def test_is_timeseries_returns_the_correct_response(
+        self,
+        metric_group: str,
+        expected_return_value: bool,
+    ):
+        """
+        Given a valid metric with either a `Headline` or `Timeseries`
+            metric group
+        When the `is_timeseries` property is called
+        Then the expected response is returned
+        """
+        # Given
+        expected_metric_group = expected_return_value
+
+        # When
+        returned_metric_group = DataSourceFileType[metric_group].is_timeseries
+
+        # Then
+        assert returned_metric_group == expected_metric_group
+
+
+class TestExtractMetricGroup:
+    @pytest.mark.parametrize(
+        "metric, metric_group",
+        (
+            [
+                ("COVID-19_headline_vaccines_spring24Uptake", "headline"),
+                ("COVID-19_headline_cases_7DayPercentChange", "headline"),
+                ("COVID-19_cases_casesByDay", "cases"),
+                ("RSV_testing_positivityByWeek", "testing"),
+                ("COVID-19_deaths_ONSRegByWeek", "deaths"),
+                ("COVID-19_healthcare_occupiedBedsRollingMean", "healthcare"),
+                ("COVID-19_vaccinations_autumn23_uptakeByDay", "vaccinations"),
+                ("non-existent-metric", None),
```

Same comment as before, returning None feels a little opaque. We should raise an error instead so its clear to the caller of this function what has happened